modules/test/helpers/common.go

```diff
@@ -19,17 +19,32 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/go-logr/logr"
 	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"k8s.io/apimachinery/pkg/types"
 
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type conditionsGetter interface {
+	GetConditions(name types.NamespacedName) condition.Conditions
+}
+
+type ConditionGetterFunc func(name types.NamespacedName) condition.Conditions
+
+// ConditionGetterFunc -
+func (f ConditionGetterFunc) GetConditions(name types.NamespacedName) condition.Conditions {
```

Comment:
with this, nova tests refactored can pass.
I cen see other operators use GetConditions() quite extensively, I believe this should help with testing expected conditions there as well